### ISSUE TITLE
fundingmanager, peer: lncli openchannel stuck

### ISSUE THREAD

**Issue by AndrewSamokhvalov** [lightningnetwork/lnd#1254]
### Background
I was trying to create channel with remote lightning network node (which also is controlled by me). I connected to the node with `lncli connect`, and it starts syncing the topology. Than I tried to open the channel with remote node, but `lncli` stuck (I haven't used `--block` flag).

It seems like `lncli` stuck because funding process wasn't fully finished. `fundingmanager` haven't sent update to `rpcserver` subsystem. The reason why `fundingmanager` haven't sent it seems to be in the fact that my node haven't received `MsgAcceptChannel` from remote node.

Guess:
* When remote node is syncing it is not able to proceed the incoming message and kick off the funding process.

The interesting part of that: When my node sent the funding error to remote node after 10m timeout (informing that it is no longer want to continue channel funding process) remote node received the error, wasn't able to find the channel, BUT LATER REMOTE NODE fundingmanager said
```
Recv'd fundingRequest(amt=0.001 BTC, push=0 mSAT, delay=144, pendingId=6b7677c7f2dcc934f189fed7f1c23ff3ae50c7cbe9f934e48e3ae930ef97db5e) from peer(03f17d338cdaba5d8dfad48b0708ce030dfa88b673b1eaa48113e8fc774873e560)
```
As if this message stuck somewhere. Also you could see that the time difference between `Receiving the MsgOpenChannel` and `Recv'd fundingRequest` is 15 minutes.

Local node (full log at the end):
```
➜  ~ grep -R 6b7677c7f2dcc934f189fed7f1c23ff3ae50c7cbe9f934e48e3ae930ef97db5e /tmp/kek.txt
/tmp/kek.txt:2018-05-15 18:16:25.658 [INF] FNDG: Target commit tx sat/kw for pendingID(6b7677c7f2dcc934f189fed7f1c23ff3ae50c7cbe9f934e48e3ae930ef97db5e): 250
/tmp/kek.txt:2018-05-15 18:16:25.658 [INF] FNDG: Starting funding workflow with 207.154.210.222:9735 for pendingID(6b7677c7f2dcc934f189fed7f1c23ff3ae50c7cbe9f934e48e3ae930ef97db5e)
/tmp/kek.txt:2018-05-15 18:16:25.658 [DBG] PEER: Sending MsgOpenChannel(temp_chan_id=6b7677c7f2dcc934f189fed7f1c23ff3ae50c7cbe9f934e48e3ae930ef97db5e, chain=000000000933ea01ad0ee984209779baaec3ced90fa3f408719526f8d77f4943, csv=144, amt=0.001 BTC, push_amt=0 mSAT, reserve=0.00001 BTC, flags=1) to 207.154.210.222:9735
/tmp/kek.txt:2018-05-15 18:27:06.453 [WRN] FNDG: reservation timed out waiting for peer (peerID:&{<nil> 61529006174472939650823495405219008016947708062089228897528487294733504679900 87249590735393797859838778138787554270873701498482215004222168480448674533757}, chanID:6b7677c7f2dcc934f189fed7f1c23ff3ae50c7cbe9f934e48e3ae930ef97db5e)
/tmp/kek.txt:2018-05-15 18:27:06.453 [ERR] FNDG: Failing funding flow: reservation timed out waiting for peer (peerID:&{<nil> 61529006174472939650823495405219008016947708062089228897528487294733504679900 87249590735393797859838778138787554270873701498482215004222168480448674533757}, chanID:6b7677c7f2dcc934f189fed7f1c23ff3ae50c7cbe9f934e48e3ae930ef97db5e) ((*lnwire.Error)(0xc430544f00)({
/tmp/kek.txt: ChanID: (lnwire.ChannelID) (len=32 cap=32) 6b7677c7f2dcc934f189fed7f1c23ff3ae50c7cbe9f934e48e3ae930ef97db5e,
/tmp/kek.txt:2018-05-15 18:27:06.453 [INF] FNDG: Cancelling funding reservation for node_key=0388082ef17afb126db9a0877e5f8a8c1a88aec86107fd1f26c2503c384c570fdc, chan_id=6b7677c7f2dcc934f189fed7f1c23ff3ae50c7cbe9f934e48e3ae930ef97db5e
/tmp/kek.txt:2018-05-15 18:27:06.454 [DBG] PEER: Sending Error(chan_id=6b7677c7f2dcc934f189fed7f1c23ff3ae50c7cbe9f934e48e3ae930ef97db5e, err=funding failed due to internal error) to 207.154.210.222:9735
/tmp/kek.txt:2018-05-15 18:30:31.630 [DBG] PEER: Received MsgAcceptChannel(temp_chan_id=6b7677c7f2dcc934f189fed7f1c23ff3ae50c7cbe9f934e48e3ae930ef97db5e, reserve=0.00001 BTC, csv=144, num_confs=3) from 207.154.210.222:9735
```

Remote node (full log at the end):
```
root@testnet:/mnt/blockchains/lightning/logs/bitcoin/testnet# grep -R 6b7677c7f2dcc934f189fed7f1c23ff3ae50c7cbe9f934e48e3ae930ef97db5e .
./lnd.log.743:2018-05-15 15:16:25.894 [DBG] PEER: Received MsgOpenChannel(temp_chan_id=6b7677c7f2dcc934f189fed7f1c23ff3ae50c7cbe9f934e48e3ae930ef97db5e, chain=000000000933ea01ad0ee984209779baaec3ced90fa3f408719526f8d77f4943, csv=144, amt=0.001 BTC, push_amt=0 mSAT, reserve=0.00001 BTC, flags=1) from 138.68.89.84:52290
./lnd.log.744:2018-05-15 15:27:06.690 [DBG] PEER: Received Error(chan_id=6b7677c7f2dcc934f189fed7f1c23ff3ae50c7cbe9f934e48e3ae930ef97db5e, err=funding failed due to internal error) from 138.68.89.84:52290
./lnd.log.744: ChanID: (lnwire.ChannelID) (len=32 cap=32) 6b7677c7f2dcc934f189fed7f1c23ff3ae50c7cbe9f934e48e3ae930ef97db5e,
./lnd.log.744:2018-05-15 15:27:06.692 [TRC] PEER: Update stream for ChannelID(6b7677c7f2dcc934f189fed7f1c23ff3ae50c7cbe9f934e48e3ae930ef97db5e) created
./lnd.log.744:2018-05-15 15:27:06.692 [ERR] PEER: recv'd update for unknown channel 6b7677c7f2dcc934f189fed7f1c23ff3ae50c7cbe9f934e48e3ae930ef97db5e from 138.68.89.84:52290
./lnd.log.744:2018-05-15 15:30:31.111 [INF] FNDG: Recv'd fundingRequest(amt=0.001 BTC, push=0 mSAT, delay=144, pendingId=6b7677c7f2dcc934f189fed7f1c23ff3ae50c7cbe9f934e48e3ae930ef97db5e) from peer(03f17d338cdaba5d8dfad48b0708ce030dfa88b673b1eaa48113e8fc774873e560)
./lnd.log.744:2018-05-15 15:30:31.572 [INF] FNDG: Requiring 3 confirmations for pendingChan(6b7677c7f2dcc934f189fed7f1c23ff3ae50c7cbe9f934e48e3ae930ef97db5e): amt=0.001 BTC, push_amt=0 mSAT
./lnd.log.744:2018-05-15 15:30:31.572 [INF] FNDG: Sending fundingResp for pendingID(6b7677c7f2dcc934f189fed7f1c23ff3ae50c7cbe9f934e48e3ae930ef97db5e)
./lnd.log.744:2018-05-15 15:30:31.572 [DBG] PEER: Sending MsgAcceptChannel(temp_chan_id=6b7677c7f2dcc934f189fed7f1c23ff3ae50c7cbe9f934e48e3ae930ef97db5e, reserve=0.00001 BTC, csv=144, num_confs=3) to 138.68.89.84:52290
```

### Your environment
* version of `lnd`
local: `bdf3f4d7753ce950bd14ed56ad96c12097f1b4c9`
remote: `321cc69e4e93ae5a176e6649702362bd2b7e38e8`
* which operating system (`uname -a` on *Nix)

local:
```
15.6.0 Darwin Kernel Version 15.6.0: Tue Jan  9 20:12:05 PST 2018; root:xnu-3248.73.5~1/RELEASE_X86_64 x86_64
```
remote:
```
4.4.0-119-generic #143-Ubuntu SMP Mon Apr 2 16:08:24 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```
* version of `btcd`, `bitcoind`, or other backend

local: 
`bitcoind=Bitcoin Core Daemon version v0.16.0.0-g4b4d7eb`
remote:
`btcd=a03db407e40d3b66ea29984263bbc8bf4d2f04c4`

### Steps to reproduce
Connect to remote node and try to open the channel while syncing the topology. But it seems that remote node should be connected to other nodes to make it hard for it to proceed messages.

[remote.2.txt](https://github.com/lightningnetwork/lnd/files/2008616/remote.2.txt)
[remote.1.txt](https://github.com/lightningnetwork/lnd/files/2008617/remote.1.txt)
[local.txt](https://github.com/lightningnetwork/lnd/files/2008618/local.txt)

**Comment by Roasbeef** [comment]
The other side cancelled back the funding request. If you attempt to start the funding process while the remote party is still validating the graph, it's natural that due to the increased load, it would prioritize processing the messages (as the RPC will be getting hammered) over attempting to finalize a funding flow.

**Comment by Roasbeef** [comment]
The issue is here:
> /tmp/kek.txt:2018-05-15 18:27:06.453 [WRN] FNDG: reservation timed out waiting for peer (peerID:&{<nil> 61529006174472939650823495405219008016947708062089228897528487294733504679900 87249590735393797859838778138787554270873701498482215004222168480448674533757}, chanID:6b7677c7f2dcc934f189fed7f1c23ff3ae50c7cbe9f934e48e3ae930ef97db5e)

The zombie sweeper times out the channel, but then doesn't send an error on the channel that is piped back to the gRPC client. Easy fix.

**Comment by cfromknecht** [comment]
@Roasbeef will look into fix for 0.5
